cmd/bitrot.go

```diff
@@ -195,3 +198,44 @@ func bitrotVerify(r io.Reader, wantSize, partSize int64, algo BitrotAlgorithm, w
 	}
 	return nil
 }
+
+// bitrotSelfTest performs a self-test to ensure that bitrot
+// algorithms compute correct checksums. If any algorithm
+// produces an incorrect checksum it fails with a hard error.
+//
+// bitrotSelfTest tries to catch any issue in the bitrot implementation
+// early instead of silently corrupting data.
+func bitrotSelfTest() {
+	var checksums = map[BitrotAlgorithm]string{
+		SHA256:          "a7677ff19e0182e4d52e3a3db727804abc82a5818749336369552e54b838b004",
+		BLAKE2b512:      "e519b7d84b1c3c917985f544773a35cf265dcab10948be3550320d156bab612124a5ae2ae5a8c73c0eea360f68b0e28136f26e858756dbfe7375a7389f26c669",
+		HighwayHash256:  "39c0407ed3f01b18d22c85db4aeff11e060ca5f43131b0126731ca197cd42313",
+		HighwayHash256S: "39c0407ed3f01b18d22c85db4aeff11e060ca5f43131b0126731ca197cd42313",
+	}
+	for algorithm := range bitrotAlgorithms {
+		if !algorithm.Available() {
+			continue
+		}
+
+		checksum, err := hex.DecodeString(checksums[algorithm])
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "bitrot: failed to decode %v checksum %s for selftest: %v", algorithm, checksums[algorithm], err)
+			os.Exit(1)
```

Comment:
can we not use logger.Fatal() @aead ?

Reply:
Didi it like this: https://github.com/minio/minio/pull/11918/commits/411ee7e4373cc86a0c30dd0875d4b1b13d16c818 -adds an error we can share.

Reply:
@harshavardhana We can - but this will not end the process. Is that not desired?!
@klauspost Thanks ❤️ - will re-use that

Reply:
logger.Fatal exits the process @aead